app/Services/PirepService.php

```diff
@@ -719,4 +726,103 @@ public function setPilotState(User $pilot, Pirep $pirep)
 
         event(new UserStatsChanged($pilot, 'airport', $previous_airport));
     }
+
+    public function handleDiversion(Pirep $pirep): void
+    {
+        // Return if diversion handling is disabled
+        if (!setting('pireps.handle_diversion', false)) {
+            return;
+        }
+
+        $diversion_airport_id = $pirep->fields->where('slug', 'diversion-airport')->first()?->value;
+
+        // Return if no diversion
+        if (!$diversion_airport_id) {
+            return;
+        }
+
+        $diversion_airport = $this->airportRepo->findWithoutFail($diversion_airport_id);
+
+        // Return if diversion airport not found and airport lookup is disabled
+        if (!$diversion_airport && !setting('general.auto_airport_lookup', false)) {
+            return;
+        }
+
+        if (!$diversion_airport) {
+            $diversion_airport = $this->airportSvc->lookupAirportIfNotFound($diversion_airport_id);
+        }
+
+        // Return if we still not have any diversion airport
+        if (!$diversion_airport) {
+            return;
+        }
+
+        $pirep->loadMissing('aircraft', 'flight', 'user');
+        $aircraft = $pirep->aircraft;
+        $flight = $pirep->flight;
+        $user = $pirep->user;
+
+        event(new PirepDiverted($pirep));
+
+        $has_vmsacars = Module::find('VMSAcars');
+
+        if ($has_vmsacars && $flight) {
+            $free_flights_enabled = DB::table('vmsacars_config')->find('free_flights_airline_aircraft_only')?->value;
```

Comment:
How does this behave if that table doesn't exist?

Reply:
I believe the `Module::find('VMSAcars')` returns either an instance of the Module or null. So if the module is not enabled then it will be null. But if you move it to the module, we don't need this check anymore since the module will definitely be enabled.